lib/active_model_serializers/test/schema.rb

```diff
@@ -10,19 +10,38 @@ module Schema
       #   get :index
       #   assert_response_schema
       def assert_response_schema(schema_path = nil, message = nil)
-        matcher = AssertResponseSchema.new(schema_path, response, message)
+        matcher = AssertResponseSchema.new(schema_path, request, response, message)
+        assert(matcher.call, matcher.message)
+      end
+
+      def assert_request_schema(schema_path = nil, message = nil)
+        matcher = AssertRequestSchema.new(schema_path, request, response, message)
+        assert(matcher.call, matcher.message)
+      end
+
+      # May be renamed
+      def assert_request_response_schema(schema_path = nil, message = nil)
+        assert_request_schema(schema_path, message)
+        assert_response_schema(schema_path, message)
+      end
+
+      def assert_schema(payload, schema_path = nil, message = nil)
+        matcher = AssertSchema.new(schema_path, request, response, message, payload)
         assert(matcher.call, matcher.message)
       end
 
       MissingSchema = Class.new(Minitest::Assertion)
       InvalidSchemaError = Class.new(Minitest::Assertion)
 
-      class AssertResponseSchema
-        attr_reader :schema_path, :response, :message
+      class AssertSchema
+        attr_reader :schema_path, :request, :response, :message, :payload
 
-        def initialize(schema_path, response, message)
+        # Interface may change.
+        def initialize(schema_path, request, response, message, payload = nil)
           require_json_schema!
+          @request = request
           @response = response
+          @payload = payload
```

Comment:
line 53 should probably be changed, `response_body` should be replaced with `@payload`

Reply:
@maksar PRs welcome.. you just commented on a PR that was closed four years ago